File: allbeingSort.ts
function isSorted(array: number[]): boolean {
    for (let i = 1; i < array.length; i++) {
      if (array[i] < array[i - 1]) {
        return false;
      }
    }
    return true;
  }
  
  function generatePermutations(array: number[], level: number, results: number[][]): void {
    if (level === array.length) {
      results.push([...array]);
      return;
    }
  
    for (let i = level; i < array.length; i++) {
      [array[level], array[i]] = [array[i], array[level]];
      generatePermutations(array, level + 1, results);
      [array[level], array[i]] = [array[i], array[level]];
    }
  }
  
  function allbeingSort(array: number[]): void {
    const permutations: number[][] = [];
    generatePermutations(array, 0, permutations);
  
    for (const permutation of permutations) {
      if (isSorted(permutation)) {
        console.log('Sorted array:', permutation);
        return;
      }
    }
  
    console.log('Array is not sortable.');
  }
  
  //示例用法
  const array: number[] = [3, 2, 1];
  
  console.log('Original array:', array);
  allbeingSort(array);
  

File: allbeingSort.rs
fn is_sorted(array: &[i32]) -> bool {
    for i in 1..array.len() {
        if array[i] < array[i - 1] {
            return false;
        }
    }
    true
}

fn generate_permutations(array: &mut [i32], level: usize, results: &mut Vec<Vec<i32>>) {
    if level == array.len() {
        results.push(array.to_vec());
        return;
    }

    for i in level..array.len() {
        array.swap(level, i);
        generate_permutations(array, level + 1, results);
        array.swap(level, i);
    }
}

fn allbeing_sort(array: &mut [i32]) {
    let mut permutations = Vec::new();
    generate_permutations(array, 0, &mut permutations);

    for permutation in permutations {
        if is_sorted(&permutation) {
            println!("Sorted array: {:?}", permutation);
            return;
        }
    }

    println!("Array is not sortable.");
}

fn main() {
    // 示例用法
    let mut array = vec![3, 2, 1];

    println!("Original array: {:?}", array);
    allbeing_sort(&mut array);
}
